libs/langchain/langchain/document_loaders/parsers/pdf_docai.py

```diff
@@ -0,0 +1,217 @@
+            And when all operations are finished, you can use their results:
+            >>> operations = parser.operations_from_names(operation_names)
+            >>> results = parser.get_results(operations)
+            >>> docs = parser.batch_parse_from_results(results)
+        """
+        operations = self.docai_parse(blobs, gcs_output_path=gcs_output_path)
+        is_running = self.is_running(operations)
+        while is_running:
+            is_running = self.is_running(operations)
+            if not is_running:
+                break
+            time.sleep(60)
+            print(".", end="")
+
+        results = self.get_results(operations=operations)
+        yield from self.batch_parse_from_results(results)
+
+    def batch_parse_from_results(
+        self, results: List[DocAIParsingResults]
+    ) -> Iterator[Document]:
+        for result in results:
+            for doc in self.parse_from_result(result):
+                yield doc
+
+    def parse_from_result(self, result: DocAIParsingResults) -> List[Document]:
+        """Creates a list of Documents from Google DocAI parsing results."""
+        from google.cloud.documentai_toolbox.wrappers.document import _get_shards
+        from google.cloud.documentai_toolbox.wrappers.page import _text_from_layout
+
+        output_gcs = result.parsed_path.split("/")
+        gcs_bucket_name = output_gcs[2]
+        gcs_prefix = "/".join(output_gcs[3:]) + "/"
+        shards = _get_shards(gcs_bucket_name, gcs_prefix)
+        docs, i = [], 0
+        for shard in shards:
+            for page in shard.pages:
+                docs.append(
+                    Document(
+                        page_content=_text_from_layout(page.layout, shard.text),
+                        metadata={"page": i + 1, "source": result.source_path},
+                    )
+                )
+                i += 1
+        return docs
+
+    @property
+    def client(self) -> "DocumentProcessorServiceClient":
+        """Returns a docai client to be used by the parser."""
+        from google.cloud.documentai import DocumentProcessorServiceClient
+
+        return DocumentProcessorServiceClient(client_options=self._options)
+
+    def operations_from_names(self, operation_names: List[str]) -> List[Operation]:
+        """Initializes Long-Running Operations from their names."""
+        from google.longrunning.operations_pb2 import (
+            GetOperationRequest,  # type: ignore
+        )
+
+        operations = []
+        for name in operation_names:
+            request = GetOperationRequest(name=name)
+            operations.append(self.client.get_operation(request=request))
+        return operations
+
+    def is_running(self, operations: List[Operation]) -> bool:
+        for op in operations:
+            if not op.done:  # type: ignore[truthy-function]
+                return False
+        return True
+
+    def docai_parse(self, blobs: List[Blob], gcs_output_path: str) -> List[Operation]:
```

Comment:
Could we copy the contents of _docai_parse here and use  ```from langchain.utils.iter import batch_iterate````

Reply:
Done